src-opam/dockerfile_opam.ml

```diff
@@ -244,7 +244,10 @@ let gen_opam2_distro ?(clone_opam_repo=true) ?arch ?labels d =
      yum_opam2 ?labels ?arch ~yum_workaround ~enable_powertools d ()
   | `Zypper -> zypper_opam2 ?labels ?arch d ()
   | `Pacman -> pacman_opam2 ?labels ?arch d ()
-  | `Cygwin -> cygwin_opam2 ?labels ?arch d ()
+  | `Cygwin ->
+     match arch with
+     | None -> failwith "Windows requires an explicit architecture"
```

Comment:
Does it need an explicit arch? If so, this should be `invalid_arg`, not `failwith`

Reply:
iirc it's for the `ocaml-env` script. Although it claims to be able to detect 64 or 32 bits platforms, I've had cases where it defaulted to 32 bits and wouldn't load the correct compiler. So in this case, I'm requiring an explicit arch to be able to pass `--32` or `--64` later on.
I'll change to `invalid_arg`.